### PR TITLE
Use EAS managed build number, run build/submit on GH Actions

### PR DESCRIPTION
**Pull request by haileyok**
This creates a better workflow for building and deploying. Also does automatic nightly TestFlights for iOS

- Use GitHub runners for deployments instead of EAS. iOS runs on `macos-14` and Android runs on `ubuntu-latest`.
- Use EAS's remove build number instead of saving locally to `app.config.js`. This eliminates the need for committing to the repo each time we deploy to TestFlight.
- The iOS job runs either manually or automatically each night at 9PM PST. The Android job only runs manually.
- Both jobs automatically deploy to the store after building. (Build times for both platforms average 15 minutes)
- On both platforms we obtain the Sentry auth tokens from a GitHub secret. On Android `google-services.json` is also obtained from another GitHub secret.

### CODE REVIEW

File: app.config.js [hunk 1]
```diff
@@ -169,7 +149,7 @@ module.exports = function () {
               organization: 'blueskyweb',
               project: 'react-native',
               release: VERSION,
-              dist: `${PLATFORM}.${VERSION}.${DIST_BUILD_NUMBER}`,
+              dist: `${PLATFORM}.${VERSION}.${new Date().toISOString()}`,
```
Comment (estrattonbailey) [inline app.config.js:152]: Sorry none of this has comments on it, but we need to keep these values in sync with our Sentry config in order for source maps to work. Let's chat in Slack 👍

**Review by estrattonbailey — CHANGES_REQUESTED** [review]
Small thing, should be easy to resolve

File: app.config.js (hunk 1 above)
Comment (haileyok) [reply app.config.js]: I think this will cover it. Going to download the IPA and install to make sure it sets the version correctly.

Comment (haileyok) [reply app.config.js]: Bundle looks good, just check Sentry and we should be good to go.

**Review by estrattonbailey — APPROVED** [review]
LGTM let's give it a go 🎉

**Pull request merged**